src/workerd/jsg/struct.h

```diff
@@ -103,8 +103,9 @@ class StructWrapper<Self, T, TypeTuple<FieldWrappers...>, kj::_::Indexes<indices
                      kj::isSameType<typename FieldWrappers::Type, Unimplemented>()) && ...)) {
         return T{};
       }
-      jsg::throwTypeError(isolate, "Cannot initialize a dictionary with required members from an "
-                                    "undefined or null value.");
+      jsg::throwTypeError(isolate, kj::str("Cannot initialize ", typeid(T).name(),
```

Comment:
minor nit but is this going to give a user friendly name that is useful for end users or something only we can appreciate?

Reply:
Friendliness is debatable, but this is definitely something that will have a name similar to TS type.